xbf_rs/src/xbf_vec/vec.rs

```diff
@@ -1,38 +1,13 @@
-use crate::{DeserializeType, Serialize, XbfMetadata, XbfType};
+use crate::{XbfMetadata, XbfType};
 use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
 use std::io::{self, Read, Write};
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct XbfVec {
-    pub(crate) inner_type: Box<XbfMetadata>,
+    pub(crate) inner_type: XbfMetadata,
```

Comment:
So we can make a XbfVec of XbfVec's without making the borrow checker mad? sweet

Reply:
Yup! The recursive type for metadata is XbfVecMetadata, so we need a box there, but for the XbfVec itself its not recursive with respect to that field so we can totally avoid the Box.